examples/execsnoop.yaml

```diff
@@ -0,0 +1,22 @@
+metrics:
+  counters:
+    - name: process_exec_total
+      help: Count process starts
+      ring_buf: true
+      labels:
+        - name: pid
```

Comment:
I'm not sure this is a good example to have. Is there a more real world example?

Reply:
Well, I'm not sure which example should I chose as well...

In fact all libbpf-tools with perf event array output can be converted in to ring buffer, they are already doing this: https://github.com/iovisor/bcc/pull/4262

Maybe I can try with other tools in [bcc/libbpf-tools](https://github.com/iovisor/bcc/blob/master/libbpf-tools)? For example:

- [sigsnoop](https://github.com/iovisor/bcc/blob/master/libbpf-tools/sigsnoop.bpf.c)
- [opensnoop](https://github.com/iovisor/bcc/blob/master/libbpf-tools/opensnoop.bpf.c)
- [mountsnoop](https://github.com/iovisor/bcc/blob/master/libbpf-tools/mountsnoop.bpf.c)
- [statsnoop](https://github.com/iovisor/bcc/blob/master/libbpf-tools/statsnoop.bpf.c)

Which one do you think is better?

Reply:
Or maybe remove the `pid` and `ppid` will make the result from execsnoop more meaningful?

```
# TYPE ebpf_exporter_process_exec_total counter
ebpf_exporter_process_exec_total{comm="askpass.sh",filename="/home/yunwei/.vscode-server/bin/8fa188b2b301d36553cbc9ce1b0a146ccb93351f/extensions/git/dist/askpass.sh"} 2
ebpf_exporter_process_exec_total{comm="cat",filename="/usr/bin/cat"} 58
ebpf_exporter_process_exec_total{comm="cpuUsage.sh",filename="/home/yunwei/.vscode-server/bin/8fa188b2b301d36553cbc9ce1b0a146ccb93351f/out/vs/base/node/cpuUsage.sh"} 12
ebpf_exporter_process_exec_total{comm="curl",filename="/usr/bin/curl"} 3
ebpf_exporter_process_exec_total{comm="git",filename="/usr/bin/git"} 22
ebpf_exporter_process_exec_total{comm="git",filename="/usr/lib/git-core/git"} 5
ebpf_exporter_process_exec_total{comm="git-remote-http",filename="/usr/lib/git-core/git-remote-https"} 2
ebpf_exporter_process_exec_total{comm="mktemp",filename="/usr/bin/mktemp"} 2
ebpf_exporter_process_exec_total{comm="node",filename="/home/yunwei/.vscode-server/bin/8fa188b2b301d36553cbc9ce1b0a146ccb93351f/node"} 3
ebpf_exporter_process_exec_total{comm="ps",filename="/usr/bin/ps"} 12
ebpf_exporter_process_exec_total{comm="rm",filename="/usr/bin/rm"} 2
ebpf_exporter_process_exec_total{comm="sed",filename="/usr/bin/sed"} 24
```

It will shows the most frequently used commands or processed in the system